### DISCUSSION TITLE
Cow questions

### DISCUSSION THREAD

**Discussion by jmmullins** [jeremyckahn/farmhand#432] [Q&A] [answered]
General questions: 

1. How many cow names are in the system by default? I've seen it offering the same names
2. Are there any plans for cows to have horns in the future or after a certain age? (It's possible I haven't gotten far enough to see this happen, so if you already have this then ignore me)
3. Is the sex only relevant for breeding, or does it make a difference for your expected milk production rates? (e.g. if I have 5 males and 1 female does only the female produce milk? Would a female-heavy herd produce more milk?)
4. At what age does the cow's value become equal to their weight? Is it 100 days? It doesn't seem like it's tied to a weight benchmark, so just curious. If you prefer to keep the logic a mystery for players to work out themselves, that's fine too. 
5. I see the "Offer" option, and the "cows traded" stat on the stats page - is there any concept of an NPC trade, or is this specifically only for players who are playing Online and there are other active players? 
Sub-question: If I'm the only active player online in the global room and I put a cow up for offer, does it notify anyone when they join the global room later on or is that offer request just lost in the void? If someone else was offering a cow, how/where would I see that notification?
6. I don't need spoilers necessarily with fine details, but is the 30-cow pen large enough to hold all the permutations of cows available in the game, or will I have to pick/choose at some point? Gotta catch 'em all!

**Comment by jeremyckahn** [comment]
> 1\. How many cow names are in the system by default? I've seen it offering the same names

The cow names come from https://github.com/jeremyckahn/fun-animal-names. There are 235 animal names and 103 adjectives, so there are are 24,205 possible combinations.

> 2\. Are there any plans for cows to have horns in the future or after a certain age? (It's possible I haven't gotten far enough to see this happen, so if you already have this then ignore me)

There are no plans for horns, but I think it would be a nice way to bring the cows to life a bit more!

> 3\. Is the sex only relevant for breeding, or does it make a difference for your expected milk production rates? (e.g. if I have 5 males and 1 female does only the female produce milk? Would a female-heavy herd produce more milk?)

Only females produce milk. Males will produce fertilizer. 💩 So, a predominantly female pen would naturally produce more milk. 🙂

> 4\. At what age does the cow's value become equal to their weight? Is it 100 days? It doesn't seem like it's tied to a weight benchmark, so just curious. If you prefer to keep the logic a mystery for players to work out themselves, that's fine too.

This is the algorithm for determining a cow's value: https://github.com/jeremyckahn/farmhand/blob/7323dfd9a5c7e34ec2eeef3ce012a37f4a39faec/src/utils/index.js#L624-L643

In plain English, this works out to:

- If cow is being offered for sale (e.g. you're looking to buy it), the value is `(cow weight) x 1.5`.
- If you own the cow, the value is `(cow weight) x (cow age)` up until the cow is 100 days old, at which point it stops increasing in value.

> 5\. I see the "Offer" option, and the "cows traded" stat on the stats page - is there any concept of an NPC trade, or is this specifically only for players who are playing Online and there are other active players?

The only cow trades in Farmhand right now are player-to-player. An NPC trade is a very interesting idea that I haven't thought of though. I like it!

> Sub-question: If I'm the only active player online in the global room and I put a cow up for offer, does it notify anyone when they join the global room later on or is that offer request just lost in the void? If someone else was offering a cow, how/where would I see that notification?

There are no notifications for available trades, but I think that would be a great UX improvement. Good thinking!

> 6\. I don't need spoilers necessarily with fine details, but is the 30-cow pen large enough to hold all the permutations of cows available in the game, or will I have to pick/choose at some point? Gotta catch 'em all!

No, there's not enough space in the 30-cow pen to have all the possible varieties unfortunately. There are 7 non-Rainbow colors and 7 style variants, so 50 total varieties overall (Rainbow Cows only have one style). It wouldn't be hard to add support for 50-cow pens, but I think it might perform poorly at runtime. 😕

-----------------------------

It's great to see Farmhand inspiring so many questions! Let me know if you have any more. 🙂

**Reply by jmmullins** [reply]
>The cow names come from https://github.com/jeremyckahn/fun-animal-names. There are 235 animal names and 103 adjectives, so there are are 24,205 possible combinations.

I don't see any of the cow names in these lists. These look like the randomly generated online player names I've seen so far though. 

>Only females produce milk

Cool, nice authentic touch. I was hoping this was the case, but hadn't experimented enough with my herd yet, thanks for the info!

>There are no notifications for available trades, but I think that would be a great UX improvement. Good thinking!

Is there anywhere I would see that someone is offering a trade? Like how would I accept?

>It wouldn't be hard to add support for 50-cow pens, but I think it might perform poorly at runtime. 😕

What if we let the user have 2 different "pages" of cow pens, i.e. pen A and pen B (east pasture vs west pasture or something). It feels like that could warrant adding a concept of "transferring" cows between pens, but I'm not sure that would even be critical, more just for visual grouping of certain cows if you want. Although maybe we would need that if you want to try breeding certain cows from different pens (unless we have the breeding pen concept allow cows to be added from either pen)

Thanks for all the info!

**Reply by jeremyckahn** [reply, marked as answer]
> I don't see any of the cow names in these lists. These look like the randomly generated online player names I've seen so far though.

Ah you're right, I was thinking of the random online player names. Sorry for the confusion. Cow names come from here: https://github.com/jeremyckahn/farmhand/blob/7323dfd9a5c7e34ec2eeef3ce012a37f4a39faec/src/data/fruit-names.js#L2-L95

There are 92 fruit/cow names in that list.

> Is there anywhere I would see that someone is offering a trade? Like how would I accept?

You can see cows being offered for trade in the "Active players" view. From there you can just click the "Trade" button for a cow (assuming you are offering a cow yourself) and it's done. Trade acceptance happens automatically. So if you offer a cow up for trade and leave it for a while, you may come back to a new cow! 🐮

> What if we let the user have 2 different "pages" of cow pens, i.e. pen A and pen B (east pasture vs west pasture or something). It feels like that could warrant adding a concept of "transferring" cows between pens, but I'm not sure that would even be critical, more just for visual grouping of certain cows if you want. Although maybe we would need that if you want to try breeding certain cows from different pens (unless we have the breeding pen concept allow cows to be added from either pen)

Supporting multiple pens would avoid the performance issues of a larger pen. Good idea! That would invite its own UX/design complexity, but it's certainly something that could be explored.

**Reply by jmmullins** [reply]
thanks @jeremyckahn!!

**Comment by jeremyckahn** [comment]
> * If I'm the only active player online in the global room and I put a cow up for offer, does it notify anyone when they join the global room later on or is that offer request just lost in the void? If someone else was offering a cow, how/where would I see that notification?

This is being added in #434.
